Rofi pass

seylerius opened the pull request:
### Description

This is a configuration module for the `rofi-pass` package. It's pretty basic at this point, only providing an option to set your password store root, and add extra configuration through the `extraConfig` option. 

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [X] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/rycee/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/rycee/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [X] Added myself and the module files to `.github/CODEOWNERS`.

teto commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,32 @@
+  meta.maintainers = [ maintainers.seylerius ];
+
+  options.programs.rofi.pass = {
+    enable = mkEnableOption "Enable rofi integration with password-store";
```
dont need the `Enable` since mkEnableOption prepends it

seylerius replied:
Thanks for the tip, fixed in 27f66d3.

teto commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,32 @@
+
+  options.programs.rofi.pass = {
+    enable = mkEnableOption "Enable rofi integration with password-store";
+    root = mkOption {
```
root is confusing ? would rename it to `stores`.

seylerius replied:
Good idea, fixed in d588a19.

teto commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,32 @@
+    extraConfig = mkOption {
+      type = types.str;
+      default = "";
+      description =
```
add an example. Can you use a types.attrs with a lib.generators.toINI instead ?

seylerius replied:
Not sure. The reason I originally didn't is that rofi-pass wants a pure-shell config file, not an INI, and I don't know how that will mesh.

rycee replied:
Even if it has to remain as a textual `extraConfig` it would be good to add an example. Just a line or two is enough.

yurrriq commented:
Any blockers here? I use both home-manager and rofi-pass, and would love to see these merge.

rycee commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,32 @@
+      type = types.str;
+      default = "";
+      description =
+        "Directory root of your password-store. Separate multiple roots with `:`, as `/foo/bar/baz:/path/to/root2`.";
```
Keep lines at around 80 characters max.

rycee replied:
Also use docbook markup, e.g., using `<literal>`.

rycee commented on tests/default.nix:
```diff
@@ -67,6 +67,7 @@ import nmt {
     ./modules/programs/zplug
     ./modules/programs/zsh
     ./modules/xresources
+
```
```suggestion
```

rycee commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,32 @@
+
+  options.programs.rofi.pass = {
+    enable = mkEnableOption "rofi integration with password-store";
+    stores = mkOption {
```
Separate options by an empty line.
```suggestion

    stores = mkOption {
```

rycee commented on .github/CODEOWNERS:
```diff
@@ -195,3 +195,6 @@
 /modules/xsession.nix                                 @rycee
+
+/modules/programs/rofi-pass.nix                      @seylerius
+/tests/modules/programs/rofi-pass/*                  @seylerius
```
The `*` is unnecessary
```suggestion
/tests/modules/programs/rofi-pass                    @seylerius
```

Can also move the block to underneath `/modules/programs/powerline-go.nix` to maintain the alphabetic order.

rycee commented:
Had a quick look and added a few comments.

seylerius commented:
Okay, I've finally come back to this, rebased it freshly on top of master, and made the requested changes. I also updated the tests to reflect an earlier change.

teto commented on tests/default.nix:
```diff
@@ -1,4 +1,4 @@
-{ pkgs ? import <nixpkgs> {} }:
+{ pkgs ? import <nixpkgs> { } }:
```
zhy the refor,qtting of this file ?

rycee replied:
I think what @teto was trying to say was "why the reformatting of this file?" 😉

Basically, avoid running nixfmt manually, simply run the format script in the project root.

rycee replied:
This file is still having an undesired major reformatting. The only change that should be here is the addition of the `./modules/programs/rofi-pass` line.

rycee commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,43 @@
+    enable = mkEnableOption "rofi integration with password-store";
+
+    stores = mkOption {
+      type = types.str;
+      default = "";
+      description = "Directory root of your password-store. "
+        + "Separate multiple roots with <literal>:</literal>, "
+        + "as <literal>/foo/bar/baz:/path/to/root2</literal>.";
```
Since multiple roots can be given it makes sense to use a list here. Also, it would be good to use regular multi-line strings in the description to fit the style in the other modules.

```suggestion
      type = types.listOf types.str;
      default = [ ];
      description = ''
        Directory roots of your password-stores.
      '';
```

rycee commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,43 @@
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
```
```suggestion
      type = types.lines;
```

rycee commented on modules/programs/rofi-pass.nix:
```diff
@@ -0,0 +1,43 @@
+        USERNAME_field='user'
+        AUTOTYPE_field='autotype'
+      '';
+      description =
```
Use multiline string.

seylerius commented:
Okay, I've made the changes y'all suggested and pushed them. Thanks muchly to you both, @rycee and @teto. My bad for forgetting about the included format script.

seylerius commented:
Did a fresh rebase on top of master to make sure it would merge cleanly.

seylerius commented:
Fixed that. Turned off my local formatting, checked back out the version from master, and added the one line back in to refer to the added tests.

seylerius commented:
Thanks for your patience and helping me with cleaning it up. Do we want to add a news item?

rycee commented:
No need to add a news item, the person merging will do that. I'll have a
look and probably merge after work today.

Sable Seyler <notifications@github.com> writes:
> Thanks for your patience and helping me with cleaning it up. Do we want to add a news item?
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/nix-community/home-manager/pull/1427#issuecomment-748010059

rycee commented:
Merged to master now. Thanks for the contribution!

yurrriq commented:
Any chance we can get this backported to release-20.09?

rycee commented:
Should be in the 20.09 branch now. Hopefully it works OK 🙂

yurrriq commented:
Thanks. After a `nix flake update` (tracking release-20-09), it's working as expected for me.
